srcts/src/shiny/bind.ts

```diff
@@ -204,10 +223,17 @@ function bindInputs(
       if (el.hasAttribute("data-shiny-no-bind-input")) continue;
       const id = binding.getId(el);
 
-      // Check if ID is falsy, skip
+      // Don't bind if ID is falsy...
       if (!id) continue;
 
-      bindingsRegistery.addBinding(id, "input");
+      // ...or if this element is already registered with a binding
+      if ($(el).data("shiny-input-binding")) {
```

Comment:
I think `el.classList.contains("shiny-bound-input")` is a better test. If I'm reading this correctly, `.shiny-bound-input` directly reflects whether this exact element is bound or not; whereas `$(el).data("shiny-input-binding")` will be truthy if this element has _ever_ been bound.

Reply:
And if you replace that test then you can simply `continue` without needing to conditionally call `bindingsRegistry.addBinding(id, "input")`.

Reply:
Yeah I went this way because I knew it'd induce conversation about which approach is better. 

`unbindOutputs()` should remove both the class and the data, so the presence of either should in theory be a solid signal. I'm inclined to lean toward the data value, since it's less likely a user would have included `data-shiny-input-binding` in the markup (but maybe they've added `.shiny-bound-input` to the class list for some reason?). If we really really want to be certain we could test the value of `$(el).data("shiny-input-binding")` but that seems excessive.

https://github.com/rstudio/shiny/blob/3e00632052b1401eb31c220c7f03c7da619b69ff/error-console-insert-ui/srcts/src/shiny/bind.ts#L404-L405

Reply:
Oh wait! Yes `unbindInputs()` leaves the data attribute in tact. So there is a currently correct answer of testing the class

Reply:
I updated `bindInputs()` to check the class in 26e51092c3a33c070df728ef86847053a7e3f606 and then removed the `.isRegistered()` method from `bindingsRegistry` since it isn't used anywhere else in b67b0623e9f8cb65175a89a369f2191e6779b838.